java/ql/lib/ext/java.lang.model.yml

```diff
@@ -37,13 +37,23 @@ extensions:
       - ["java.lang", "CharSequence", True, "charAt", "", "", "Argument[-1]", "ReturnValue", "taint", "manual"]
       - ["java.lang", "CharSequence", True, "subSequence", "", "", "Argument[-1]", "ReturnValue", "taint", "manual"]
       - ["java.lang", "CharSequence", True, "toString", "", "", "Argument[-1]", "ReturnValue", "taint", "manual"]
+      - ["java.lang", "IllegalArgumentException", False, "IllegalArgumentException", "(String)", "", "Argument[0]", "Argument[-1].SyntheticField[java.lang.IllegalArgumentException.message]", "value", "manual"]
+      - ["java.lang", "IllegalArgumentException", False, "getMessage", "()", "", "Argument[-1].SyntheticField[java.lang.IllegalArgumentException.message]", "ReturnValue", "value", "manual"]
```

Comment:
All of these `getMessage()` methods are overrides of the one on `Throwable`. Note that we would like the following to have flow:
```
Exception e = new RuntimeException("source");
sink(e.getMessage());
```
so we need all of these to use the same synthetic field - which makes sense, since the actual implementation uses just one message field on Throwable. That means that we can have just one model for `getMessage` on Throwable, which should then also apply to overrides.

You may want to add a test along the lines of the above - then you can see that it currently fails and check that it succeeds when you have updated the models.

Reply:
Thanks! Updated in https://github.com/github/codeql/pull/11572/commits/08546549bf8585ebcf87899bbd2d140315c8f154.